x-pack/plugin/eql/src/test/resources/querytranslator_tests.txt

```diff
@@ -841,7 +841,7 @@ disjunctionOfMultiFieldsFunctions
 process where concat(file_name, ".", process_name) == "foo" or add(pid, ppid) > 100
 ;
 {"bool":{"should":[{"script":{"script":{"source":"InternalEqlScriptUtils.multiValueDocValues(doc,params.v0,X0->InternalEqlScriptUtils.multiValueDocValues(doc,params.v1,X1->InternalQlScriptUtils.nullSafeFilter(InternalQlScriptUtils.eq(InternalEqlScriptUtils.concat([X0,params.v2,X1]),params.v3))))"
-"params":{"v0":"file_name","v1":"process_name","v2":".","v3":"foo"}}
+"params":{"v0":"file_name.keyword","v1":"process_name","v2":".","v3":"foo"}}
```

Comment:
This change raises some questions:
- why `v0` has this change while `v1` doesn't? Both are parameters of `InternalEqlScriptUtils.multiValueDocValues` script method
- what field types are `file_name` and `process_name`? Is one `text` (the `file_name`) and the other `keyword` (the `process_name`) and this explains why the change applies to only one of them?
- how come no one complained about this behavior so far? The multi-value behavior is the default for few versions now, and I was expecting this to hit at least one user. Is everyone using just `keyword` fields or is there some other explanation to why this worked fine so far?

Reply:
Your guess is correct, `file_name` is `text`, while `process_name` is `keyword` (see https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/eql/src/test/resources/mapping-default.json)

About last question, unfortunately I don't have an answer. My guess is that, even though many operators are affected, the actual use cases are quite uncommon (eg. equality in the projections or usage of scalar functions in GROUP BY). Another explanation could be that the work-around is easy to guess, eg. a query like 

```
SELECT LEAST(a, b) x FROM ...
```

can easily be rewritten as 

```
SELECT LEAST(a.keyword, b.keyword) x FROM ...
```

so probably it's easier for the end user to adapt to the limitation than to file a new issue